addons/delivery/models/product_packaging.py

```diff
@@ -9,19 +9,23 @@ class ProductPackaging(models.Model):
 
 
     def _get_default_length_uom(self):
-        return self.env['product.template']._get_length_uom_name_from_ir_config_parameter()
+        return self.env['product.template']._get_length_uom_id_from_ir_config_parameter()
 
     def _get_default_weight_uom(self):
-        return self.env['product.template']._get_weight_uom_name_from_ir_config_parameter()
+        return self.env['product.template']._get_weight_uom_id_from_ir_config_parameter()
 
     height = fields.Integer('Height')
     width = fields.Integer('Width')
     packaging_length = fields.Integer('Length')
     max_weight = fields.Float('Max Weight', help='Maximum weight shippable in this packaging')
     shipper_package_code = fields.Char('Package Code')
     package_carrier_type = fields.Selection([('none', 'No carrier integration')], string='Carrier', default='none')
-    weight_uom_name = fields.Char(string='Weight unit of measure label', compute='_compute_weight_uom_name', default=_get_default_weight_uom)
-    length_uom_name = fields.Char(string='Length unit of measure label', compute='_compute_length_uom_name', default=_get_default_length_uom)
+    weight_uom_id = fields.Many2one(
```

Comment:
```suggestion
    weight_uom_name = fields.Char(related="weight_uom_id.display_name")
    length_uom_name = fields.Char(related="length_uom_id.display_name")
    weight_uom_id = fields.Many2one(
```
For the sake of retrocompatibility :wink: